### PR TITLE
Add constref to various class-methods

### PR DESCRIPTION
**Pull request by WarPie**
added `constref` "modifier" to a bunch of exported class-methods, notably the once often used through `client`. Also added a couple to `TMufasaBitmap`, maybe more should be added, I only added to the most obvious methods.

Might be other classes that could benefit from `constref` but this is the most important ones I think.

With this I got github.com/SRL/SRL to compile after some other modifications (to account for other lape changes)

### CODE REVIEW

File: Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas [hunk 1]
```diff
@@ -512,25 +512,25 @@ procedure Register_TMufasaBitmap(Compiler: TLapeCompiler);
...
+    addGlobalFunc('procedure TMufasaBitmap.ValidatePoint(x,y : integer); constref;', @TMufasaBitmap_ValidatePoint);
     addGlobalFunc('function TMufasaBitmap.SaveToFile(const FileName : string): boolean;', @TMufasaBitmap_SaveToFile);
     addGlobalFunc('procedure TMufasaBitmap.LoadFromFile(const FileName : string);', @TMufasaBitmap_LoadFromFile);
     addGlobalFunc('procedure TMufasaBitmap.Rectangle(const Box : TBox;FillCol : TColor);', @TMufasaBitmap_Rectangle);
     addGlobalFunc('procedure TMufasaBitmap.Rectangle(const Box: TBox; const Color: Integer; const Transparency: Extended); overload;', @TMufasaBitmap_RectangleEx);
     addGlobalFunc('procedure TMufasaBitmap.FloodFill(const StartPT : TPoint; const SearchCol, ReplaceCol : TColor);', @TMufasaBitmap_FloodFill);
-    addGlobalFunc('procedure TMufasaBitmap.SetPixel(x,y : integer; Color : TColor);', @TMufasaBitmap_FastSetPixel);
-    addGlobalFunc('procedure TMufasaBitmap.SetPixels(Points : TPointArray; Colors : TIntegerArray);', @TMufasaBitmap_FastSetPixels);
+    addGlobalFunc('procedure TMufasaBitmap.SetPixel(x,y : integer; Color : TColor); constref;', @TMufasaBitmap_FastSetPixel);
```
Comment (nielsAD) [inline Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas]: Any reason why you change `SetPixel`, but not, for example, `DrawClear`?

File: Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas [hunk 2]
```diff
@@ -512,25 +512,25 @@ procedure Register_TMufasaBitmap(Compiler: TLapeCompiler);
     addClassVar('TMufasaBitmap', 'Height', 'Integer', @TMufasaBitmap_Height_Read);
     addGlobalFunc('procedure TMufasaBitmap.SetPersistentMemory(mem: PtrUInt; awidth, aheight: integer);', @TMufasaBitmap_SetPersistentMemory);
     addGlobalFunc('procedure TMufasaBitmap.ResetPersistentMemory();', @TMufasaBitmap_ResetPersistentMemory);
-    addGlobalFunc('function TMufasaBitmap.PointInBitmap(x,y : integer): boolean;', @TMufasaBitmap_PointInBitmap);
-    addGlobalFunc('procedure TMufasaBitmap.ValidatePoint(x,y : integer);', @TMufasaBitmap_ValidatePoint);
+    addGlobalFunc('function TMufasaBitmap.PointInBitmap(x,y : integer): boolean; constref;', @TMufasaBitmap_PointInBitmap);
+    addGlobalFunc('procedure TMufasaBitmap.ValidatePoint(x,y : integer); constref;', @TMufasaBitmap_ValidatePoint);
     addGlobalFunc('function TMufasaBitmap.SaveToFile(const FileName : string): boolean;', @TMufasaBitmap_SaveToFile);
```
Comment (nielsAD) [inline Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas]: Prime example of a `constref` function, in my opinion.

Comment (nielsAD) [inline Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas]: Prime example of a `constref` function, in my opinion.

**Comment by nielsAD** [comment]
I think all FPC/Delphi classes have methods with `constref` characteristics: they use `Self`, but never modify it. Maybe it's an idea to have type classes `constref` by default and have them marked as mutable instead.

File: Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas (hunk 1 above)
Comment (WarPie) [inline Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas]: I was very quick when going over TMufasaBitmap
"*Also added a couple to TMufasaBitmap, maybe more should be added, I only added to the most obvious methods.*"
The most obvious methods for me was the ones that I instantly recognized.

File: Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas (hunk 2 above)
Comment (WarPie) [inline Units/MMLAddon/LPInc/Classes/MML/lptmufasabitmap.pas]: Yeah, I was VERY quick with TMufasaBitmap, just took the ones that I instantly recognized.

**Comment by WarPie** [comment]
I can agree with that suggestion, should be harmless to do that.

**Comment by JohnPeel** [comment]
Do it.

**Comment by nielsAD** [comment]
Let's continue in #398.

**Pull request closed without merging**